### PR TITLE
[Core] Add functions to judge whether the channel/objlist exists or not 

### PR DESCRIPTION
**Pull request by Gyingguo**
[Core][Channel] Add has_channel function in core/channel/channel_factory_base.hpp
[Core][Objlist] Add has_objlist function in core/objlist_factory.hpp
ref #69

### CODE REVIEW

File: core/channel/channel_factory_base.hpp [hunk 1]
```diff
@@ -117,6 +117,10 @@ class ChannelFactoryBase {
         channel_map.erase(name);
     }
 
+    static bool has_channel(const std::string& name = "") {
+        return !(channel_map.find(name) == channel_map.end());
```
Comment (Yuzhen11) [inline core/channel/channel_factory_base.hpp]: What about `return channel_map.find(name) != channel_map.end()` ?

Comment (zzxx-husky) [inline core/channel/channel_factory_base.hpp]: And why we need a default value here?

Comment (kygx-legend) [inline core/channel/channel_factory_base.hpp]: @Gyingguo take actions?

Comment (Gyingguo) [inline core/channel/channel_factory_base.hpp]: ok

**Comment by kygx-legend** [comment]
Could you run `./scripts/clang-format.py -o replace` and then push again?

**Comment by kygx-legend** [comment]
LGTM.

**Pull request merged**